sql/catalyst/src/main/java/org/apache/spark/sql/util/CaseInsensitiveStringMap.java

```diff
@@ -157,4 +163,11 @@ public double getDouble(String key, double defaultValue) {
     String value = get(key);
     return value == null ? defaultValue : Double.parseDouble(value);
   }
+
+  /**
+   * Returns the original case-sensitive map.
+   */
+  public Map<String, String> getOriginalMap() {
```

Comment:
What about `asCaseSensitiveMap`? That is more clear that using "original" because that doesn't indicate why someone would call this method.